cmd/bosun/sched/check.go

```diff
@@ -322,22 +324,38 @@ func (s *Schedule) CollectStates() {
 func (r *RunHistory) GetUnknownAndUnevaluatedAlertKeys(alert string) (unknown, uneval []expr.AlertKey) {
 	unknown = []expr.AlertKey{}
 	uneval = []expr.AlertKey{}
+	anyFound := false
 	for ak, ev := range r.Events {
 		if ak.Name() != alert {
 			continue
 		}
+		anyFound = true
 		if ev.Status == StUnknown {
 			unknown = append(unknown, ak)
 		} else if ev.Unevaluated {
 			uneval = append(uneval, ak)
 		}
 	}
+	if !anyFound {
```

Comment:
I don't understand this logic. It first checks to see if there were any events in the run history for the given alertkey. If there were not, then it checks the global state. Why is it like this? Hangout is perhaps best to explain.